### PR TITLE
test framework: remove native mode

### PR DESCRIPTION
**Pull request by howardjohn**
Motivation and info:  https://docs.google.com/document/d/18QgpvBH9N8Io5xU-0piysyOYif65U03m8GabpvHb4IQ/edit#bookmark=id.d0fk1kp4f1ok

Note: this does not clean up *all* references to environment. Followup cleanup PRs will be needed as well

### CODE REVIEW

**Comment by kyessenov** [comment]
If there is no native environment, there's just k8s and why do we need environment complexity?

**Comment by howardjohn** [comment]
> If there is no native environment, there's just k8s and why do we need environment complexity?

Sorry forgot to put context link. We are removing the "environment" concept completed. I updated the PR description

File: pkg/test/framework/components/istio/operator.go [hunk 1]
```diff
@@ -92,7 +92,6 @@ var leaderElectionConfigMaps = []string{
 	leaderelection.IngressController,
 	leaderelection.NamespaceController,
 	leaderelection.ValidationController,
-	leaderelection.StatusController,
```
Comment (stevenctl) [inline pkg/test/framework/components/istio/operator.go:95]: This change doesn't seem related?

Comment (howardjohn) [reply pkg/test/framework/components/istio/operator.go:95]: Good catch.. yes its not I got a bit ambitious on this and fixed another issue (https://github.com/istio/istio/issues/24524) but should probably split it out

File: tests/integration/pilot/sidecar_api_test.go [hunk 2]
```diff
@@ -60,24 +57,16 @@ func TestSidecarListeners(t *testing.T) {
 					return true, nil
 				})
 
-			// TODO: The code below is flaky. We should re-enable this once we have explicit config loading trigger support in Galley.
+			namespace.ClaimOrFail(ctx, ctx, "seexamples")
```
Comment (lei-tang) [inline tests/integration/pilot/sidecar_api_test.go:60]: Is "seexamples" a typo? Should it be "see-examples" instead?

File: tests/integration/README.md [hunk 3]
```diff
@@ -432,36 +430,14 @@ If you need to add a new test suite, it can be added to the [job configuration](
...
-```bash
-sudo sysctl net.ipv4.conf.all.forwarding=1
-sudo iptables -P FORWARD ACCEPT
-# On some machines, an additional rule may be needed to allow traffic from all `br-...` docker bridge interfaces
-sudo iptables -A INPUT -i br-+ -j ACCEPT
-```
-
-### Kubernetes Environment
-
 The test binaries run in a Kubernetes cluster, but the test logic runs in the test binary. To specify the Kubernetes
```
Comment (lei-tang) [inline tests/integration/README.md:433]: The sentence "To specify the Kubernetes environment" can be removed because the PR removes the "--istio.test.env kube" parameter in the line followed by this sentence.

**Comment by niwatorrri** [comment]
A reminder that you might want to update the wiki page [Istio Test Framework](https://github.com/istio/istio/wiki/Istio-Test-Framework)
